sympy/core/power.py

```diff
@@ -1735,9 +1735,13 @@ def mul(d1, d2):
                 raise NotImplementedError()
         if not d.is_positive:
             g = g.simplify()
-            _, d = g.leadterm(x)
```

Comment:
`d` should be recomputed when it is not zero as it may be positive for the simplified `g`. This line should not be removed. It should come before the test below (line 1739 1740).